sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/AbstractInstrumentBuilder.java

```diff
@@ -61,50 +67,52 @@ public BuilderT setDescription(String description) {
     return getThis();
   }
 
-  private InstrumentDescriptor makeDescriptor(InstrumentType type, InstrumentValueType valueType) {
-    return InstrumentDescriptor.create(instrumentName, description, unit, type, valueType);
-  }
-
   protected <T> T swapBuilder(SwapBuilder<T> swapper) {
     return swapper.newBuilder(
         meterProviderSharedState, meterSharedState, instrumentName, description, unit);
   }
 
   final <I extends AbstractInstrument> I buildSynchronousInstrument(
-      InstrumentType type,
-      InstrumentValueType valueType,
       BiFunction<InstrumentDescriptor, WriteableMetricStorage, I> instrumentFactory) {
-    InstrumentDescriptor descriptor = makeDescriptor(type, valueType);
+    InstrumentDescriptor descriptor =
+        InstrumentDescriptor.create(instrumentName, description, unit, type, valueType);
     WriteableMetricStorage storage =
         meterSharedState.registerSynchronousMetricStorage(descriptor, meterProviderSharedState);
     return instrumentFactory.apply(descriptor, storage);
   }
 
-  final CallbackRegistration registerDoubleAsynchronousInstrument(
+  final SdkObservableInstrument registerDoubleAsynchronousInstrument(
       InstrumentType type, Consumer<ObservableDoubleMeasurement> updater) {
```

Comment:
isn't the type available now at the class level, and hence not needed to be passed in?

Reply:
oh, I think I see... for types that can be both observable and not, we need to be able to pass this in, correct?

Reply:
Yes exactly. The type is assumed to be the synchronous version (i.e. COUNTER, UP_DOWN_COUNTER, etc) until a callback is regsitered.